### PR TITLE
[gh-1625] fix creating session keys from session keys.

### LINKED ISSUE #1625: [bug] SessionKey can create SessionKey
https://github.com/rooch-network/rooch/blob/417f226614f56caa4e8657a067d9bf952567530f/frameworks/rooch-framework/sources/session_key.move#L116-L117

The assert ensures users can not use a SessionKey to create a new SessionKey, but it does not work. 

## Reason

https://github.com/rooch-network/rooch/blob/417f226614f56caa4e8657a067d9bf952567530f/crates/rooch-types/src/framework/auth_validator.rs#L152-L164

the `transaction_validtor` of above should `transaction_validator`.

1. All the logic with `TxValidateResult` should be checked.
2. Write a test to ensure the assert.

### PR DESCRIPTION
**Pull request by feliciss**
## Summary

1. Add test cases for creating session keys from session keys.
2. Check params and logic within `TxValidateResult`.

- Closes #1625

### CODE REVIEW

File: crates/testsuite/features/cmd.feature [hunk 1]
```diff
@@ -37,10 +37,15 @@ Feature: Rooch CLI integration tests
...
 
       # session key
-      Then cmd: "session-key create  --app-name test --app-url https:://test.rooch.network --scope 0x3::empty::empty"
+      Then cmd: "session-key create --app-name test --app-url https://test-seed.rooch.network --scope 0x3::empty::empty"
       Then cmd: "move run --function 0x3::empty::empty  --session-key {{$.session-key[-1].authentication_key}}"
       Then assert: "{{$.move[-1].execution_info.status.type}} == executed"
 
+      # session key from session key
+      Then cmd: "session-key create --app-name test --app-url https://test-seed.rooch.network --session-key {{$.session-key[-1].authentication_key}} --scope 0x3::empty::empty"
+      Then cmd: "move run --function 0x3:cargo test -p testsuite --test integration:empty::empty  --session-key {{$.session-key[-1].authentication_key}}"
```
Comment (jolestar) [inline crates/testsuite/features/cmd.feature:46]: can not run testsuite in testsuite case.

**Comment by jolestar** [comment]
How about this PR? We plan to release a new version.

File: crates/testsuite/features/cmd.feature (hunk 1 above)
Comment (feliciss) [reply crates/testsuite/features/cmd.feature]: Will remove it.

**Comment by feliciss** [comment]
> How is this PR going? We plan to release a new version.

Pushed a new hash for review. Still on-going.

**Comment by jolestar** [comment]
https://discord.com/channels/1078938449974935592/1237388145893445672

**Comment by feliciss** [comment]
> @feliciss discuss in https://discord.com/channels/1078938449974935592/1237388145893445672

I don't have permissions to send messages to that thread.

**Pull request closed without merging**